pkg/virt-handler/cgroup/cgroup_v1_manager.go

```diff
@@ -109,3 +113,92 @@ func getCurrentlyDefinedRules(runcManager runc_cgroups.Manager) ([]*devices.Rule
 func (v *v1Manager) GetCpuSet() (string, error) {
 	return getCpuSetPath(v, "cpuset.cpus")
 }
+
+func rw_filecontents(freadpath string, fwritepath string) (err error) {
+	rfile, err := os.Open(freadpath)
+	if err != nil {
+		return fmt.Errorf("Open failed: %s (%v)", freadpath, err)
+	}
+	defer rfile.Close()
+
+	wfile, err := os.OpenFile(fwritepath, os.O_RDWR, 0755)
+	if err != nil {
+		return fmt.Errorf("OpenFile failed: %s (%v)", fwritepath, err)
+	}
+	defer wfile.Close()
+
+	data := make([]byte, 20000)
+	count, err := rfile.Read(data)
+	if err != nil {
+		return fmt.Errorf("OpenFile failed: %s (%v)", fwritepath, err)
+	}
+
+	_, err = wfile.Write(data[:count])
+	if err != nil {
+		return fmt.Errorf("rw_filecontents: Write failed: %s", fwritepath)
+	}
+
+	return nil
+}
+
+// Attach TID to cgroup. Optionally on a subcgroup of
+// the pods control group (if subcgroup != nil).
+func (v *v1Manager) AttachTID(subsystem string, subcgroup string, tid int) error {
+	cgrouppath, err := v.GetBasePathToHostSubsystem(subsystem)
+	if err != nil {
+		return err
+	}
+	if subcgroup != "" {
+		cgrouppath = filepath.Join(cgrouppath, subcgroup)
+	}
+
+	wval := strconv.Itoa(tid)
+
+	err = runc_cgroups.WriteFile(cgrouppath, "tasks", wval)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func init_cgroup(grouppath string, newcgroupname string) (exists bool, err error) {
+	newgrouppath := filepath.Join(grouppath, newcgroupname)
+	if _, err := os.Stat(newgrouppath); !os.IsNotExist(err) {
+		return true, err
+	}
+	err = util.MkdirAllWithNosec(newgrouppath)
+	if err != nil {
+		log.Log.Infof("mkdir %s failed", newgrouppath)
+		return false, err
+	}
+	for _, fname := range []string{"cpuset.mems", "cpuset.cpus"} {
+		rpath := filepath.Join(grouppath, fname)
+		wpath := filepath.Join(newgrouppath, fname)
+
+		err = rw_filecontents(rpath, wpath)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	return false, nil
+}
+
+func (v *v1Manager) CreateChildCgroup(name string, subsystem string) (bool, error) {
+	subsystemPath := v.Path(subsystem)
+	if subsystemPath == "" {
+		return false, fmt.Errorf("controller %s does not exist", subsystem)
+	}
+	subsyspath := filepath.Join(HostCgroupBasePath, subsystemPath)
+	exists, err := init_cgroup(subsyspath, name)
+	if exists == true {
```

Comment:
shouldn't you check for the error first and then do the comparison?

Reply:
The first check of init_cgroup is for the existance of a directory, so no, not needed, thanks.